flink-table/flink-table-common/src/main/java/org/apache/flink/table/data/utils/JoinedRowData.java

```diff
@@ -16,9 +16,16 @@
  * limitations under the License.
  */
 
-package org.apache.flink.table.data;
+package org.apache.flink.table.data.utils;
```

Comment:
I would suggest to move this class to `org.apache.flink.table.data`, because I think it is a kind of `RowData` which is similar to `GenericRowData`, not a utility.

Reply:
I don't have strong opinion here but it is marked `@Internal` and is not a core `RowData` class. It has not even a proper hashCode/equals. To me this is rather a utility class.

Reply:
I don't have a strong opinion as well. My argument of keeping the class in `org.apache.flink.table.data` was maintaining backwards-compatability, but based on the established guarantees and expectations this does not seem essential.

I will add a hotfix commit which adds `hashCode` and `equals` before changing the package. Feel free to keep or drop the package change commit when you are merging the PR.